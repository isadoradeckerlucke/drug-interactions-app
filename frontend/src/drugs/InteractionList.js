import React from "react";
import Table from "react-bootstrap/Table";

function InteractionList(interactions) {
  let interactionsObj = interactions.interactions;

  if (interactionsObj.noneFound) {
    return <p>{interactionsObj.noneFound}</p>;
  }
  let interactionsTable = [];
  Object.entries(interactionsObj).forEach(([key, value]) => {
    interactionsTable.push(
      <tr key={key}>
        <td>{value.name1}</td>
        <td>{value.name2}</td>
        <td>{value.description}</td>
        <td>{value.severity}</td>
        <td>{value.comment}</td>
      </tr>
    );
  });
  return (
    <div>
      <Table responsive striped bordered hover>
        <thead>
          <tr>
            <th>drug 1</th>
            <th>drug 2</th>
            <th>description</th>
            <th>severity</th>
            <th>comment</th>
          </tr>
        </thead>
        <tbody>{interactionsTable}</tbody>
      </Table>
    </div>
  );
}

export default InteractionList;
