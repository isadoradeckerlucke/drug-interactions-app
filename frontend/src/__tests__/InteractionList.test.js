import React from "react";
import { render } from "@testing-library/react";
import InteractionList from "../drugs/InteractionList";

describe("interaction list works as expected", function () {
  const interactionsFound = {
    interactions: {
      "00": {
        comment: "test drug is resolved to test drug",
        description: "test description",
        name1: "drug1test",
        name2: "drug2test",
        severity: "high",
      },
      "01": {
        comment: "test drug is resolved to test drug",
        description: "test description",
        name1: "drug1test2",
        name2: "drug2test2",
        severity: "N/A",
      },
    },
  };

  const interactionsNotFound = {
    interactions: {
      noneFound: "there were no interactions found with this test drug",
    },
  };
  /** smoke test */
  it("renders without crashing", function () {
    render(<InteractionList interactions={interactionsFound} />);
  });

  /** snapshot test */
  it("matches snapshot with interactions found", function () {
    const { asFragment } = render(
      <InteractionList interactions={interactionsFound} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with no interactions found", function () {
    const { asFragment } = render(
      <InteractionList interactions={interactionsNotFound} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
