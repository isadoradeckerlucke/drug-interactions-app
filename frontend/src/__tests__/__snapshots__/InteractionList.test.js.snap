// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interaction list works as expected matches snapshot with interactions found 1`] = `
<DocumentFragment>
  <div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              drug 1
            </th>
            <th>
              drug 2
            </th>
            <th>
              description
            </th>
            <th>
              severity
            </th>
            <th>
              comment
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td />
            <td />
            <td />
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`interaction list works as expected matches snapshot with no interactions found 1`] = `
<DocumentFragment>
  <div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              drug 1
            </th>
            <th>
              drug 2
            </th>
            <th>
              description
            </th>
            <th>
              severity
            </th>
            <th>
              comment
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td />
            <td />
            <td />
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
